### PR TITLE
Better documentations

### PR DESCRIPTION
**Pull request by aschnapp**
### Overview
Just readme and documentation updates.

### Includes:
See files tab for reference

### Developer Checklist:
- [x] The code follows the Quality Policy file on Google Drive
- [x] My code has been developer-tested and includes unit tests
- [x] I have considered proper use of exceptions
- [x] I have eliminated IDE warnings

### Notes:
NA

### Refs:
- [HM-319](https://the-halo-mod.atlassian.net/browse/HM-319)
- #89 
- #91 
- #92

### CODE REVIEW

**Review by vperuzzi — APPROVED** [review]

**Comment by aneuhold** [comment]
@aschnapp is this still a work in progress?

**Comment by aschnapp** [comment]
@aneuhold mostly done, one last thing which is still getting resolved in slack

**Comment by aschnapp** [comment]
@aneuhold this is done now.

File: server/README.md [hunk 1]
```diff
@@ -70,3 +68,37 @@ The `server` is the back-end to TheHaloMod-SPA application. It provides the API,
...
+```
+Testing
+```sh
+python3 -m pytest;;
+```
+Linting
+```sh
+autopep8 -a -r --in-place halomod_app # fixes autofixable lint errors
+flake8 tests halomod_app;;
+```
```
Comment (aneuhold) [inline server/README.md:99]: It looks like the new sections need to be added to the table of contents for this readme.

**Review by aneuhold — CHANGES_REQUESTED** [review]
Looks great! Just needs one comment addressed and it is good.

**Comment by aschnapp** [comment]
@aneuhold done!

**Review by aneuhold — APPROVED** [review]
Looks good!

**Pull request merged**